controllers/chat/initChatFromPost.js

```diff
@@ -54,39 +54,57 @@ const initChatFromPost = async (req, res) => {
 
     //Add anon info to channel
     if (targetUserModel.is_anonymous) {
-      const postAnonUserInfo = await PostAnonUserInfoFunction.checkSelfUsernameInPost(
-        PostAnonUserInfo,
-        User,
-        {
-          postId: postId,
-          userId: targetUserModel.user_id
-        },
-        null,
-        true
-      );
+      let anon_user_info = {};
+      if (source === 'post') {
+        const postAnonUserInfo = await PostAnonUserInfoFunction.checkSelfUsernameInPost(
+          PostAnonUserInfo,
+          User,
+          {
+            postId: postId,
+            userId: targetUserModel.user_id
+          },
+          null,
+          true
+        );
+
+        anon_user_info = {
+          anon_user_info_color_code: postAnonUserInfo?.anon_user_info_color_code,
+          anon_user_info_color_name: postAnonUserInfo?.anon_user_info_color_name,
+          anon_user_info_emoji_code: postAnonUserInfo?.anon_user_info_emoji_code,
+          anon_user_info_emoji_name: postAnonUserInfo?.anon_user_info_emoji_name
+        };
+      } else if (source === 'comment') {
+        const reaction = await Getstream.feed.getReactionById(commentId);
+        anon_user_info = {
+          anon_user_info_color_code: reaction?.data.anon_user_info_color_code,
+          anon_user_info_color_name: reaction?.data.anon_user_info_color_name,
+          anon_user_info_emoji_code: reaction?.data.anon_user_info_emoji_code,
+          anon_user_info_emoji_name: reaction?.data.anon_user_info_emoji_name
+        };
+      }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The code for fetching anonymous user information based on the source (either 'post' or 'comment') is repeated twice with minor differences. This can be refactored into a separate function to improve maintainability and readability. Here's how you could do it:

```diff
+ async function fetchAnonUserInfo(source, postId, commentId, targetUserId) {
+   let anon_user_info = {};
+   if (source === 'post') {
+     const postAnonUserInfo = await PostAnonUserInfoFunction.checkSelfUsernameInPost(
+       PostAnonUserInfo,
+       User,
+       {
+         postId: postId,
+         userId: targetUserId
+       },
+       null,
+       true
+     );
+     anon_user_info = {
+       anon_user_info_color_code: postAnonUserInfo?.anon_user_info_color_code,
+       anon_user_info_color_name: postAnonUserInfo?.anon_user_info_color_name,
+       anon_user_info_emoji_code: postAnonUserInfo?.anon_user_info_emoji_code,
+       anon_user_info_emoji_name: postAnonUserInfo?.anon_user_info_emoji_name
+     };
+   } else if (source === 'comment') {
+     const reaction = await Getstream.feed.getReactionById(commentId);
+     anon_user_info = {
+       anon_user_info_color_code: reaction?.data.anon_user_info_color_code,
+       anon_user_info_color_name: reaction?.data.anon_user_info_color_name,
+       anon_user_info_emoji_code: reaction?.data.anon_user_info_emoji_code,
+       anon_user_info_emoji_name: reaction?.data.anon_user_info_emoji_name
+     };
+   }
+   return anon_user_info;
+ }

- let anon_user_info = {};
- if (source === 'post') {
-   const postAnonUserInfo = await PostAnonUserInfoFunction.checkSelfUsernameInPost(
-     PostAnonUserInfo,
-     User,
-     {
-       postId: postId,
-       userId: targetUserModel.user_id
-     },
-     null,
-     true
-   );
-   anon_user_info = {
-     anon_user_info_color_code: postAnonUserInfo?.anon_user_info_color_code,
-     anon_user_info_color_name: postAnonUserInfo?.anon_user_info_color_name,
-     anon_user_info_emoji_code: postAnonUserInfo?.anon_user_info_emoji_code,
-     anon_user_info_emoji_name: postAnonUserInfo?.anon_user_info_emoji_name
-   };
- } else if (source === 'comment') {
-   const reaction = await Getstream.feed.getReactionById(commentId);
-   anon_user_info = {
-     anon_user_info_color_code: reaction?.data.anon_user_info_color_code,
-     anon_user_info_color_name: reaction?.data.anon_user_info_color_name,
-     anon_user_info_emoji_code: reaction?.data.anon_user_info_emoji_code,
-     anon_user_info_emoji_name: reaction?.data.anon_user_info_emoji_name
-   };
- }
+ let anon_user_info = await fetchAnonUserInfo(source, postId, commentId, targetUserModel.user_id);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->